File: standing-out/StandingOut.Data/DTO/ClassSessionFeatures.cs
using System.Collections.Generic;

namespace StandingOut.Data.DTO
{
    public class ClassSessionFeatures
    {
        public int TutorDashboard_EditLesson_Session_MaxPersons { get; set; }
        public int TutorDashboard_CreateLesson_Session_MaxPersons { get; set; }
        
        // Take the below 6 switches into account when creating/editing course 
        // Private/Public Course - limit number of lessons using PrivateLessonCount/PublicLessonCount
        // Under18 Course - Allow radio as per bool
        public int TutorDashboard_CreateCourse_PrivateLessonCount { get; set; }
        public int TutorDashboard_EditCourse_PrivateLessonCount { get; set; }
        public int TutorDashboard_CreateCourse_PublicLessonCount { get; set; }
        public int TutorDashboard_EditCourse_PublicLessonCount { get; set; }
        public int Classroom_ClassroomEntryTime_MinutesBeforeEntry { get; set; }
        public bool TutorDashboard_CreateCourse_Under18 { get; set; }
        public bool TutorDashboard_EditCourse_Under18 { get; set; }

        public bool AdminDashboard_DBSApproval_ApprovalRequired { get; set; }
        public bool AdminDashboard_ProfileApproval_ApprovalRequired { get; set; }

        public int ClassroomTutorCommand_Groups_MaxGroups { get; set; }
        public int TutorDashboard_Lesson_MaxGroups { get; set; }
        public bool Menu_Panes_TriPaneEnabled { get; set; }
        public bool Menu_Panes_QuadPaneEnabled { get; set; }

        //public bool OpenDoc_Panes_TriPaneEnabled { get; set; }
        //public int TutorDashboard_Lesson_GroupSize { get; set; }
        public bool Classroom_EnterClass_SessionRecordingEnabled { get; set; }
        public bool TutorDashboard_View_CompletedLesson { get; set; }
        public bool StudentDashboard_View_CompletedLesson { get; set; }

        // public decimal[] Admin_CommissionPerStudent { get; set; }

        public List<CommissionPerStudentTier> Admin_CommissionPerStudent_StudentAttendancePerMonthTiers { get; set; } 
            = new List<CommissionPerStudentTier>();
    }

    public class CommissionPerStudentTier
    {
        public string RuleCriteria { get; set; }
        public int RuleMin { get; set; }
        public int RuleMax { get; set; }
        public decimal Setting { get; set; }
    }
}

File: standing-out/StandingOutStore/AngularSource/app/models/class-session-features.ts
export class ClassSessionFeatures {
    tutorDashboard_EditLesson_Session_MaxPersons: number
    tutorDashboard_CreateLesson_Session_MaxPersons: number
    //tutorDashboard_CreateLesson_PrivateLessonAllowed: boolean
    //tutorDashboard_EditLesson_PrivateLessonAllowed: boolean
    tutorDashboard_CreateCourse_PrivateLessonCount: number
    tutorDashboard_EditCourse_PrivateLessonCount: number
    tutorDashboard_CreateCourse_PublicLessonCount: number
    tutorDashboard_EditCourse_PublicLessonCount: number

    classroom_ClassroomEntryTime_MinutesBeforeEntry: number
    //tutorCommand_Groups_MaxGroupSize: number
    //tutorDashboard_Lesson_GroupSize: number
    classroomTutorCommand_Groups_MaxGroups: number
    tutorDashboard_Lesson_MaxGroups: number

    menu_Panes_TriPaneEnabled :boolean
    menu_Panes_QuadPaneEnabled:boolean
    //openDoc_Panes_TriPaneEnabled: boolean
    //openDoc_Panes_QuadPaneEnabled: boolean
    classroom_EnterClass_SessionRecordingEnabled: boolean
    tutorDashboard_View_CompletedLesson: boolean
    studentDashboard_View_CompletedLesson: boolean
    admin_CommissionPerStudent_StudentAttendancePerMonthTiers: any[]
    admin_CommissionPerStudent: number[]

    tutorDashboard_CreateCourse_Under18: boolean
    tutorDashboard_EditCourse_Under18: boolean
    adminDashboard_DBSApproval_ApprovalRequired: boolean
    adminDashboard_ProfileApproval_ApprovalRequired: boolean
}
    //  admin_CommissionPerStudent_StudentAttendancePerMonthTiers AN ITEM = 
    //  {  "ruleCriteria": "StudentAttendanceRange",
    //      "ruleMin": 11,
    //      "ruleMax": 20,
    //      "setting": 3.25 }

